internal/manifest/report_test.go

```diff
@@ -267,6 +281,154 @@ var reportTests = []struct {
 	add:     []sliceAndEntry{{entry: sampleDir, slice: oneSlice}},
 	mutate:  []*fsutil.Entry{&sampleDir},
 	err:     `cannot mutate path in report: /example-dir/ is a directory`,
+}, {
+	summary: "Hard link to regular file",
+	add: []sliceAndEntry{
+		{entry: sampleFile, slice: oneSlice},
+		{entry: sampleHardLinkReg, slice: oneSlice}},
+	expected: map[string]manifest.ReportEntry{
+		"/example-file": {
+			Path:       "/example-file",
+			Mode:       0777,
+			SHA256:     "example-file_hash",
+			Size:       5678,
+			Slices:     map[*setup.Slice]bool{oneSlice: true},
+			HardLinkId: 1,
+		},
+		"/example-hard-link-reg": {
+			Path:       "/example-hard-link-reg",
+			Mode:       sampleFile.Mode,
```

Comment:
There is a deeper question here of whether we want to make the test result depend on the test inputs. That is, if we want to, for example, use `sampleFile.Mode` instead of the mode. I would argue this is NOT what we should do for the following reason. There are some tests that only make sense given two modes (for example, changing the mode of a file). What happens if we later change `sampleFile.Mode`and it matches the final mode we are testing, we could then not do a chmod and we wouldn't notice. The point being that the output being manual encourages review when the inputs change